Bits AI - refresh and SRE docs (#29655)
DataDog/documentation: content/en/bits_ai/bits_ai_sre/investigate_alerts.md

---
title: Investigate Alerts
---

## Get started with alert investigations

### Enable Bits on monitors for automated investigations

There are two main ways to enable Bits for automated investigations: 
- **Option 1: Use the Bits-Enabled Monitors list**
  1. In Bits AI, go to the [**Bits-Enabled Monitors**][5] page. 
  1. In the **Monitors** tab, select one or more monitors, then click **Enable Bits AI**.
- **Option 2: Add the Bits AI tag to a monitor**
  1. In the [Monitor List][6], select one or more monitors to edit.
     - To edit one monitor, click the monitor to open it, then click **Edit**.
     - To edit multiple monitors, select them, then click **Edit tags**.
  1. Add the `bitsai:enabled` tag to your selected monitors.

You can also add the tag to your desired monitors using the Datadog API or Terraform. 

<div class="alert alert-info">Bits only supports metric, logs, APM, anomaly, forecast, integration, and outlier monitors for investigations.</div>

> ### Configure where investigation results are sent

Reviewer: Ahhh sorry, I'm rereading this section in whole and I wonder if it's clear for the readers. 

the tl;dr is:
- Users set up monitors and use the @slack / @oncall / @case handles to tell Datadog where to route notifications to. 
- For cases and pages, Bits automatically writes to those objects, no further configuration needed.
- For Slack, users would need to additionally "Go to [Bits AI > Settings > Integrations](https://app.datadoghq.com/bits-ai/settings/integrations) and connect your Slack workspace." to enable Bits to write to the monitor Slack notification that's created.

---

> title: Investigate alerts

Reviewer 2: ```suggestion
title: Investigate Alerts
```

There are two main ways to enable Bits for automated investigations: 
- **Option 1: Use the Bits-Enabled Monitors list**
  1. In Bits AI, go to the [**Bits-Enabled Monitors**][5] page. 
  1. In the **Monitors** tab, select one or more monitors, then click **Enable Bits AI**.
- **Option 2: Add the Bits AI tag to a monitor**
  1. In the [Monitor List][6], select one or more monitors to edit.
     - To edit one monitor, click the monitor to open it, then click **Edit**.
     - To edit multiple monitors, select them, then click **Edit tags**.
  1. Add the `bitsai:enabled` tag to your selected monitors.

You can also add the tag to your desired monitors using the Datadog API or Terraform. 

<div class="alert alert-info">Bits only supports metric, logs, APM, anomaly, forecast, integration, and outlier monitors for investigations.</div>

### Configure where investigation results are sent

Bits can send investigation results to several destinations. By default, results appear in two places:
  - **Full investigation results** are available on the [Bits AI Investigations][2] page.
  - **A summary of the results** is available on the status page for the monitor.

> Additionally, if you already have already configured `@slack`, `@oncall`, or `@case` [notifications in your monitor][8], Bits automatically writes to those places. If not, you can add them as destinations for investigation results to appear:

Reviewer 2: ```suggestion
Additionally, if you have already configured `@slack`, `@oncall`, or `@case` [notifications in your monitor][8], Bits automatically writes to those places. If not, you can add them as destinations for investigation results to appear:
```

### Manually start an investigation

Alternatively, you can manually invoke Bits on an individual monitor event. 

<!-- TKTK CAN'T SEE BUTTON YET -->
- **Option 1: Monitor Status Page**
  -  On the monitor status page for an alert event, click **Investigate with Bits AI**.
- **Option 2: Slack**
  - Under a monitor notification in Slack, type, `@Datadog Investigate this alert`.

For best results, see [Optimize monitors for Bits AI SRE](#optimize-monitors-for-bits-ai-sre).

### Optimize monitors for Bits AI SRE

To help Bits produce the most accurate and helpful investigation results, follow these guidelines:

- Scope the monitor to a service by either filtering the query to a specific service or grouping it by service, where appropriate. 
- Tag the monitor with a service, where appropriate. 
- Add relevant troubleshooting steps to the monitor message to give Bits a starting point. Think of the first page you'd visit in Datadog if this monitor were to fire. Consider including:
  - Plain-language instructions 
  - At least one helpful link to:
    - A Datadog dashboard
    - A logs query 
    - A trace query
    - A Datadog notebook with key graphs or instructions 

>     - A Confluence page

Reviewer: ```suggestion
```

Reviewer: removing this!

{{< img src="bits_ai/optimization_example.png" alt="Example monitor with optimization steps applied" style="width:100%;" >}}

### Restrict access to Bits investigations

To access investigations, users need the `bits_investigations_read` permission. This permission is included in the Datadog Read Only Role by default. If your organization uses custom roles, add this permission to the appropriate role. For more information on managing permissions, see [Access Control][11].

**Note**: Your organization's third-party AI enablement status is always respected, even when users have this permission. 

## How Bits AI SRE investigates

Investigations happen in two phases:

1. **Initial context gathering**

>    1. Bits begins by looking at any troubleshooting steps, Confluence pages, or Datadog links that you've added to the monitor's message—and making relevant queries. 

Reviewer 2: ```suggestion
   1. Bits begins by looking at any troubleshooting steps, Confluence pages, or Datadog links that you've added to the monitor's message, and uses them to make relevant queries.
```

| Ask for elaborations of its findings           | `Tell me more about the {issue}.`                                                  | Investigation findings               |
| Look up information about a service            | `Are there any ongoing incidents for {example-service}?`                              | Software Catalog service definitions |
| Find recent changes for a service              | `Were there any recent changes on {example-service}?`                              | Change Tracking events               |
| Find a dashboard                               | `Give me the {example-service} dashboard.`                                         | Dashboards                           |
| Query APM request, error, and duration metrics | `What's the current error rate for {example-service}?`                             | APM metrics                          |
| Search for information in Confluence           | `Find me the runbook in Confluence to rollback deployments for {example-service}.` | Confluence                           |

## Help Bits AI SRE learn

> Reviewing Bits' findings not only validates their accuracy, but also helps Bits learn from any mistakes it makes, helping it produce faster and more accurate investigations in the future.

Reviewer: ```suggestion
Reviewing Bits' findings not only validates their accuracy, but also helps Bits learn from any mistakes it makes, enabling it to produce faster and more accurate investigations in the future.
```
